### PR TITLE
Bump Dependency

### PR DESCRIPTION
**Pull request by tangledbytes**
Signed-off-by: Utkarsh Srivastava <srivastavautkarsh8097@gmail.com>

### Explain the changes
This PR is the first PR in an attempt to migrate from Node v14 to Node v16. This PR bumps patch/minor/major dependencies.

```
==========================================================
############## MAJOR DEPENDENCY UPGRADES #################
==========================================================
@google-cloud/storage    5.20.5  →    6.4.2 DONE
@types/mongodb           3.6.20  →    4.0.6 SKIPPED
@types/node            16.11.59  →  18.7.18 SKIPPED
mocha                     9.2.2  →   10.0.0 DONE - FIXED
mongodb                   3.7.3  →    4.9.1 SKIPPED
node-addon-api            4.3.0  →    5.0.0 DONE
node-gyp                  8.4.1  →    9.1.0 DONE
sinon                    13.0.2  →   14.0.0 DONE
uuid                      8.3.2  →    9.0.0 DONE
yaml                     1.10.2  →    2.1.1 DONE
==========================================================

==========================================================
########### MINOR/PATCH DEPENDENCY UPGRADES ##############
==========================================================
@azure/storage-blob              12.9.0  →   12.11.0 DONE
@google-cloud/storage            5.18.1  →    5.20.5 DONE - FIXED
@types/lodash                  4.14.182  →  4.14.185 DONE
@types/node                    16.11.33  →  16.11.59 SKIPPED
aws-sdk                        2.1127.0  →  2.1216.0 DONE
eslint                           8.14.0  →    8.23.1 DONE
ip                                1.1.5  →     1.1.8 DONE
istanbul-reports                  3.1.4  →     3.1.5 DONE
moment-timezone                  0.5.34  →    0.5.37 DONE
mongo-query-to-postgres-jsonb    0.2.14  →    0.2.15 DONE
nan                              2.15.0  →    2.16.0 DONE
pg                                8.7.3  →     8.8.0 DONE
pkg                               5.6.0  →     5.8.0 DONE
prebuild-install                  7.1.0  →     7.1.1 DONE
prom-client                      14.0.1  →    14.1.0 DONE
ts-node                          10.7.0  →    10.9.1 DONE
typescript                        4.6.4  →     4.8.3 DONE
ws                                8.6.0  →     8.8.1 DONE
==========================================================
```

**Breaking changes:**
1. `@google-cloud/storage` **_minor_** bump had a breaking change for **us**. We modified the useragent function which is internal to the module. This broke because of their code restructuring.
2. `mocha` _**major**_ bump had a breaking change. They changed their binary name from `/bin/mocha` to `/bin/mocha.js`

**Tested the above by running:**
1. `make test`
3. `make test-postgres`
4. `make tester; docker run --privileged --rm --name test1 noobaa-tester ./src/test/unit_tests/run_npm_test_on_test_container.sh -s sudo_index.js`

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,9928 @@
 {
   "name": "noobaa-core",
   "version": "5.12.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (liranmauda) [inline package-lock.json:4]: By this line, it was done using npm > 6 (7 or 8)
please revert this and use npm v6 for this (and avoid possible brakes in the DS)
We should bump npm from v6 to v8 as part of bumping node 14 to 16

**Review by liranmauda — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (utkarsh-pro) [reply package-lock.json:4]: Ah that's right, completely missed that. It was npmv8. Will revert this and will re-generate a new lock file.

**Comment by liranmauda** [comment]
@utkarsh-pro does the mocha change affect the line below? https://github.com/noobaa/noobaa-core/blob/a89b385450f4aa3096df3f05e84c566d46e047da/package.json#L21

We should probably also change in: 
https://github.com/noobaa/noobaa-core/blob/a89b385450f4aa3096df3f05e84c566d46e047da/src/deploy/NVA_build/Tests.Dockerfile#L58

**Comment by utkarsh-pro** [comment]
> @utkarsh-pro does the mocha change affect the line below?
> 
> https://github.com/noobaa/noobaa-core/blob/a89b385450f4aa3096df3f05e84c566d46e047da/package.json#L21  

Nope, this symbolic link still points to the correct module.

> We should probably also change in:
> 
> https://github.com/noobaa/noobaa-core/blob/a89b385450f4aa3096df3f05e84c566d46e047da/src/deploy/NVA_build/Tests.Dockerfile#L58

Yes, missed this. Thank you.

**Review by liranmauda — APPROVED** [review]
LGTM

**Pull request merged**